### PR TITLE
Fix CI and compatibility with Python < 3.8

### PR DESCRIPTION
**Pull request by Schnouki**
Started just to fix CI... In the end, this changes a bunch of stuff:

- changes the base image of CI to use an Ubuntu version where all wanted Python versions are available
- updates classifiers in `setup.py` to match what's tested. (3.3 and 3.4 can't be tested anymore, but are not removed from classifiers)
- partially reverts #68, which is 3.8+ only, to restore support for 2.x and <3.8
- tests more Python and PyPy versions
- fixes everything required to have that green tick ✅

### CODE REVIEW

**Comment by Schnouki** [comment]
Latest CI run in my fork: https://github.com/Schnouki/q/actions/runs/4942208141. All green 🙂

**Comment by skorokithakis** [comment]
This looks good, thank you! I have only a small comment.

File: q.py [hunk 1]
```diff
@@ -260,6 +263,35 @@ def get_call_exprs(self, caller_frame, line):
         except SyntaxError:
             return None
 
+        if self.sys.version_info >= (3, 8):
+            return self._get_accurate_call_exprs(caller_frame, line, tree)
+
+        return self._get_basic_call_exprs(caller_frame, line, tree)
+
+    def _get_basic_call_exprs(self, caller_frame, line, tree):
```
[the remark this answers is not in the document]
Comment (Schnouki) [reply q.py:271]: Sure!

**Review by skorokithakis — APPROVED** [review]

**Comment by skorokithakis** [comment]
Sorry, this has taken a while. It looks good, do you want me to merge?

**Comment by Schnouki** [comment]
No worries! And yep, definitely OK to merge it, CI is green from 2.7 to 3.12 now :tada:

**Comment by skorokithakis** [comment]
Excellent, thank you!

**Comment by skorokithakis** [comment]
By the way, @Schnouki, if you're interested, you can be a maintainer here too.

**Comment by Schnouki** [comment]
Hmmm, I think I'd like that. I use this package a lot ☺️ Please let me know what I have to do to become a co-maintainer, and I'll do my best!

**Pull request merged**